contracts/polygon/connectors/fluidity/main.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.6;
+pragma experimental ABIEncoderV2;
+
+/**
+ * @title Fluidity.
+ * @dev
+ */
+
+import { Events } from "./events.sol";
+import { Helpers } from "./helper.sol";
+import { TokenInterface } from "../../common/interfaces.sol";
+
+abstract contract FluidityResolver is Events, Helpers {
+	/**
+	 * @dev
+	 * @notice
+	 * @param token_ Token Address.
+	 * @param amt Token Amount.
+	 * @param getId ID to retrieve amt
+	 * @param setId ID stores the amount of tokens supplied
+	 */
+	function supply(
+		address token_,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	) public returns (string memory _eventName, bytes memory _eventParam) {
+		uint256 amt_ = getUint(getId, amt);
+
+		TokenInterface tokenContract = TokenInterface(token_);
+		amt_ = amt_ == type(uint256).max
+			? tokenContract.balanceOf(address(this))
+			: amt_;
+
+		Helpers.approve(tokenContract, address(this), amt_);
```

Comment:
```suggestion
		approve(tokenContract, address(protocolModule), amt_);
```
Here we are approving the funds that are in the DSA to the protocol, so the protocol can use `transferFrom` to get the funds in its address.